lib/abci/handlers/stateTransition/unserializeStateTransitionFactory.js

```diff
@@ -35,6 +36,10 @@ function unserializeStateTransitionFactory(createIsolatedDpp) {
         throw new InvalidArgumentAbciError('State Transition is invalid', { errors: e.getErrors() });
       }
 
+      if (e instanceof InvalidDocumentTypeError) {
```

Comment:
I feel like `InvalidDocumentTypeError` should be part of `InvalidStateTransitionError`. How is it possible?

Reply:
We decided to do this as a separate task.